all-actors/src/main/java/org/sunbird/cert/actor/CertificateVerifierActor.java

```diff
@@ -0,0 +1,224 @@
+        }
+        Response response = new Response();
+        response.getResult().put("response", verificationResponse);
+        sender().tell(response, getSelf());
+        logger.info("onReceive method call End");
+    }
+
+    /**
+     * Verifies signed certificate , verify signature value nad expiry date
+     *
+     * @param certificate certificate object
+     * @return
+     * @throws SignatureException.UnreachableException
+     * @throws SignatureException.VerificationException
+     */
+    private VerificationResponse verifySignedCertificate(Map<String, Object> certificate) throws SignatureException.UnreachableException, SignatureException.VerificationException {
+        List<String> messages = new ArrayList<>();
+        CollectionUtils.addIgnoreNull(messages, verifySignature(certificate));
+        CollectionUtils.addIgnoreNull(messages, verifyExpiryDate((String) certificate.get(JsonKey.EXPIRES)));
+//        messages.add(verifySignature(certificate));
+//        messages.add(verifyExpiryDate((String) certificate.get(JsonKey.EXPIRES)));
+//        messages.removeAll(Collections.singleton(null));
+        return getVerificationResponse(messages);
+    }
+
+    /**
+     * verifies the hosted certificate
+     * verifies expiry date
+     *
+     * @param certificate certificate object
+     * @return
+     */
+    private VerificationResponse verifyHostedCertificate(Map<String, Object> certificate) {
+        List<String> messages = new ArrayList<>();
+        messages.add(verifyExpiryDate((String) certificate.get(JsonKey.EXPIRES)));
+        messages.removeAll(Collections.singleton(null));
+        return getVerificationResponse(messages);
+    }
+
+    private VerificationResponse getVerificationResponse(List<String> messages) {
+        VerificationResponse verificationResponse = new VerificationResponse();
+        if (messages.size() == 0) {
+            verificationResponse.setValid(true);
+        } else {
+            verificationResponse.setValid(false);
+        }
+        verificationResponse.setErrorCount(messages.size());
+        verificationResponse.setMessages(messages);
+        return verificationResponse;
+
+    }
+
+
+    /**
+     * to download certificate from cloud
+     *
+     * @param url
+     * @return
+     * @throws IOException
+     * @throws BaseException
+     */
+    private Map<String, Object> downloadCertJson(String url) throws IOException, BaseException {
+        StoreConfig storeConfig = new StoreConfig(certsConstant.getStorageParamsFromEvn());
+        CertStoreFactory certStoreFactory = new CertStoreFactory(null);
+        ICertStore certStore = certStoreFactory.getCloudStore(storeConfig);
+        certStore.init();
+        try {
+            String uri = new URL(url).getPath();
+            String fileName = getFileName(uri);
+            certStore.get(uri.substring("/".concat(certsConstant.getSlug().concat("/")).length()));
+            File file = new File("conf/" + fileName);
```

Comment:
How can we avoid putting this logic of concatenation in multiple places? It is not good to note conf/ and concats. Also, consider renaming this as downloadCert.

Reply:
without knowing the file path, how can I read the file, so  I have to mention the file path